File: 08-1-2-State-Solution/States/SuperAngryState.cs
namespace State.States
{
    internal sealed class SuperAngryState : PartnerStateBase
    {
        private int _kissCounter;

        public SuperAngryState(Partner partner)
            : base(partner)
        {
        }

        public override void Talk()
        {
            _kissCounter = 0;
            Print("... I will never talk to you again!");
        }

        public override void Kiss()
        {
            _kissCounter++;

            if (_kissCounter < 5)
            {
                Print("...");
                return;
            }

            Print("Okay, I forgive you.");
            SetState(new NeutralState(Partner));
        }

        public override void Insult()
        {
            _kissCounter = 0;

            Print("You shall burn in hell!");
        }
    }
}

File: 09-DesignPatterns-Java/Solution/State/src/States/SuperAngryState.java
package States;

public final class SuperAngryState extends PartnerStateBase {
    private int _kissCounter;

    public SuperAngryState(Partner.Partner partner) {
        super(partner);
    }

    @Override
    public void talk() {
        _kissCounter = 0;
        print("... I will never talk to you again!");
    }

    @Override
    public void kiss() {
        _kissCounter++;

        if (_kissCounter < 5) {
            print("...");
            return;
        }

        print("Okay, I forgive you.");
        setState(new NeutralState(_partner));
    }

    @Override
    public void insult() {
        _kissCounter = 0;
        print("You shall burn in hell.");
    }
}
